ttnn/cpp/ttnn/operations/ccl/erisc_datamover_builder.cpp

```diff
@@ -0,0 +1,414 @@
+//                         |    header      |
+//             &payload->  |----------------|
+//                         |                |
+//                         |    payload     |
+//                         |                |
+//        &channel_sync->  |----------------|
+//                         |  channel_sync  |
+//                         ------------------
+//
+
+FabricEriscDatamoverConfig::FabricEriscDatamoverConfig(
+    std::size_t channel_buffer_size_bytes, std::size_t sender_ratio_size, std::size_t receiver_ratio_size) {
+    TT_ASSERT(channel_buffer_size_bytes > sizeof(tt::fabric::PacketHeader) + 2 * FabricEriscDatamoverConfig::eth_channel_sync_size);
+    const std::size_t channel_buffer_size_with_channel_sync =
+        channel_buffer_size_bytes + sizeof(tt::fabric::PacketHeader); // + 16 // sizeof(tt::fabric::PacketHeader);
+
+    this->channel_buffer_size_bytes = channel_buffer_size_bytes;
+    this->channel_buffer_size_bytes_with_channel_sync = channel_buffer_size_with_channel_sync;
+    const std::size_t total_ratio_count = 2 * sender_ratio_size + receiver_ratio_size;
+    this->sender_0_channel_size_bytes = tt::round_down(
+        (available_channel_buffering_space / total_ratio_count) * sender_ratio_size,
+        channel_buffer_size_with_channel_sync);
+    this->sender_0_num_buffers = this->sender_0_channel_size_bytes / channel_buffer_size_with_channel_sync;
+    this->sender_1_channel_size_bytes = tt::round_down(
+        (available_channel_buffering_space / total_ratio_count) * sender_ratio_size,
+        channel_buffer_size_with_channel_sync);
+    this->sender_1_num_buffers = this->sender_1_channel_size_bytes / channel_buffer_size_with_channel_sync;
+    this->receiver_channel_size_bytes = tt::round_down(
+        (available_channel_buffering_space / total_ratio_count) * receiver_ratio_size,
+        channel_buffer_size_with_channel_sync);
+    this->receiver_num_buffers = this->receiver_channel_size_bytes / channel_buffer_size_with_channel_sync;
+
+    this->sender_0_channel_base_address = buffer_region_start;
+    this->sender_1_channel_base_address = this->sender_0_channel_base_address + this->sender_0_channel_size_bytes;
+    this->receiver_channel_base_address = this->sender_1_channel_base_address + this->sender_1_channel_size_bytes;
+
+    log_trace(tt::LogOp, "Sender 0 channel_start: {}", this->sender_0_channel_base_address);
+    log_trace(tt::LogOp, "Sender 1 channel_start: {}", this->sender_1_channel_base_address);
+    log_trace(tt::LogOp, "Receiver channel_start: {}", this->receiver_channel_base_address);
+
+    TT_ASSERT(
+        this->sender_0_channel_size_bytes + this->sender_1_channel_size_bytes + this->receiver_channel_size_bytes <=
+        this->available_channel_buffering_space);
+    TT_ASSERT(
+        this->receiver_channel_base_address + this->receiver_channel_size_bytes <
+        eth_l1_mem::address_map::MAX_L1_LOADING_SIZE);
```

Comment:
is it a deliberate decision to not use TT_FATAL here?

Reply:
I typically put anything that I consider as internal logic error as asserts because we should catch them during testing - particularly anything not "user" facing. 

Though that being said, I can add a fatal on the `channel_buffer_size_bytes` argument to make sure it's not too big

Reply:
> as internal logic error as asserts because we should catch them during testing 
except that we don't run TT_ASSERTS on CI yet.

There is a way to run CI to specify RefWithDebInfo or Debug, but thats not the default behavior yet.